metriq_gym/benchmarks/benchmark.py

```diff
@@ -31,10 +31,43 @@ def from_quantum_job(cls, quantum_job, **kwargs):
         return cls(provider_job_ids=flatten_job_ids(quantum_job), **kwargs)
 
 
-class BenchmarkResult(BaseModel):
-    """Stores the final results of the benchmark"""
+class BenchmarkScore(BaseModel):
+    value: float
+    uncertainty: float = 0.0
+
 
-    pass
+class BenchmarkResult(BaseModel):
+    """Base class for benchmark results with minimal author burden.
+
+    Subclasses declare metric fields as numbers (float/int) or BenchmarkScore.
+    - Numbers map to results.values[<field>] = number and results.uncertainties[...] = 0.0
+    - BenchmarkScore maps to values[...] = value and uncertainties[...] = uncertainty
+    """
+
+    def _iter_metric_items(self):
+        for name in self.__class__.model_fields:
+            val = getattr(self, name, None)
+            if val is None:
+                continue
+            if isinstance(val, BenchmarkScore):
+                yield name, float(val.value), float(val.uncertainty)
+            elif isinstance(val, (int, float)):
+                yield name, float(val), 0.0
+            # else: ignore non-numeric / non-score fields
+
+    @property
+    def values(self) -> dict[str, float]:
+        out: dict[str, float] = {}
+        for k, v, _u in self._iter_metric_items():
+            out[k] = v
+        return out
+
+    @property
+    def uncertainties(self) -> dict[str, float]:
+        out: dict[str, float] = {}
+        for k, _v, u in self._iter_metric_items():
+            out[k] = u
+        return out
```

Comment:
These functions are both the same except a single letter. Do we want to consolidate them in some form. For instance: 

```py
from functools import cached_property

@cached_property
def _values_and_uncertainties(self) -> tuple[dict[str, float], dict[str, float]]:
    values: dict[str, float] = {}
    uncertainties: dict[str, float] = {}
    for k, v, u in self._iter_metric_items():
        values[k] = v
        uncertainties[k] = u
    return values, uncertainties

@property
def values(self) -> dict[str, float]:
    return self._values_and_uncertainties[0]

@property
def uncertainties(self) -> dict[str, float]:
    return self._values_and_uncertainties[1]
```